lib/file_service.dart

```
@@ -0,0 +1,268 @@
+        "text3": text3,
+        "text4": text4,
+        "text5": text5,
+        "text5Link": text5Link,
+        "text6": text6
+      }
+    };
+  }
+
+  @override
+  Future<void> share(
+      String message,
+      List<dynamic> titles,
+      List<dynamic> subTitles,
+      Map<String, String> texts,
+      ShareFileType saveFormat) async {
+    try {
+      // Add the generated widgets to the PDF
+      final dataForFile = await organizeDataForFile(titles, subTitles, texts);
+      Map<String, dynamic> file;
+      switch (saveFormat) {
+        case ShareFileType.PDF:
+          file = createPDF(
+              titles,
+              subTitles,
+              dataForFile["texts"]!,
+              dataForFile["image"]!,
+              dataForFile["mainTitle"]!,
+              dataForFile["ttf"]!,
+              dataForFile["PDFpageFormat"]!,
+              dataForFile["realData"]!);
+          break;
+        default:
+          file = {"file": null, "format": null};
+      }
+
+      // Save the PDF and share it
+      if (file["file"] == null || file["format"] == null) {
+        return;
+      }
+      final tempFile = await saveTemp(file["file"], file["format"]);
+      XFile tempXFile = XFile(tempFile.path);
+      await Share.shareXFiles([tempXFile], text: message);
+    } catch (e) {
+      print('Error: $e');
+    }
+  }
+
+  static Future<String?> saveAndroid(Uint8List pdfData, String format) async {
+    print(pdfData); // Print the PDF data for debugging purposes
+    try {
+      // Open a save file dialog to allow the user to select a location to save the PDF
+      String? outputFile = await FilePicker.platform.saveFile(
+        dialogTitle: 'Please select an output file:', // Dialog title
+        fileName: 'התוכנית שלי.$format', // Default file name
+        bytes: pdfData, // PDF data to be saved
+      );
+      //If the user cancels the download
+      return outputFile;
+    } catch (e) {
+      print('Error: $e');
```

why do we ignore the exception on production?